build.sh

```diff
@@ -144,33 +137,36 @@ if (( ${NUMARGS} == 0 )) || hasArg libcuml || hasArg prims || hasArg bench; then
 fi
 
 # Run all make targets at once
+
 MAKE_TARGETS=
-if (( ${NUMARGS} == 0 )) || hasArg libcuml; then
+if hasArg libcuml; then
     MAKE_TARGETS="${MAKE_TARGETS}cuml++ cuml ml ml_mg"
 fi
-if (( ${NUMARGS} == 0 )) || hasArg prims; then
+if hasArg prims; then
     MAKE_TARGETS="${MAKE_TARGETS} prims"
 fi
-if (( ${NUMARGS} == 0 )) || hasArg bench; then
+if hasArg bench; then
     MAKE_TARGETS="${MAKE_TARGETS} sg_benchmark"
 fi
 
+
+# If there are no targets specified when calling build.sh, it will
+# just call `make -j`. This avoids a lot of extra printing
+cd ${LIBCUML_BUILD_DIR}
+make -j${PARALLEL_LEVEL} ${MAKE_TARGETS} VERBOSE=${VERBOSE} ${INSTALL_TARGET}
+
 # build cumlcomms library
-if [ "${MAKE_TARGETS}" != "" ]; then
-    cd ${LIBCUML_BUILD_DIR}
-    make -j${PARALLEL_LEVEL} ${MAKE_TARGETS} VERBOSE=${VERBOSE} ${INSTALL_TARGET}
+mkdir -p ${CUML_COMMS_BUILD_DIR}
```

Comment:
question: should we only build cumlcomms if the user has NOT passed `--singlegpu` option?

Reply:
We have an interesting situation regarding the flag. In reality `single gpu` right now only disables the algorithms that require `libcumlprims` (kmeans, spmg ols and tsvd), not knn or random forest. It is mainly here to enable people to build without `libcumlprims` right now. I think a good follow up would be allowing more granularity in options to really have a single gpu build, and also being able to disable/enable algorithms. What do you think?

Reply:
or probably we can keep it simple by just saying that this flag will *only* enable *all* single-gpu algos, when specified?

Reply:
advanced users can still feel free to run the raw cmake/make commands and choose the exact ones they want.

Reply:
Sounds good. This probably will be a follow up PR to not complicate this PR then

Reply:
can you file an issue for this, please?

Reply:
yessir